src/models/translations.ts

```diff
@@ -1,18 +1,46 @@
 import Joi from 'joi';
 
+const translationStringSchema = Joi.string().allow('').required();
+const translationSchema = (example: string) => Joi.alternatives([
+  Joi.object({
+    plural: translationStringSchema,
+    singular: translationStringSchema,
+  }),
+  translationStringSchema.example(example),
+]);
 const schema = Joi.object().keys({
-  en: Joi.string().allow('').example('English string').required(),
-  nl: Joi.string().allow('').example('Nederlandse string').required(),
+  en: translationSchema('English string'),
+  nl: translationSchema('Nederlandse string'),
 })
   .tag('translations')
   .meta({ className: 'translations' });
 
-interface Translations {
-  en: string,
-  nl: string,
-}
+type SimpleTranslation = string
+type PluralizedTranslation = Record<'plural' | 'singular', string>
+type Translation = SimpleTranslation | PluralizedTranslation
+
+const isSimpleTranslation = (tr: Translation): tr is SimpleTranslation => typeof tr === 'string';
+
+const isPluralizedTranslation = (tr: Translation): tr is PluralizedTranslation => typeof tr === 'object'
+  && 'plural' in tr
+  && 'singular' in tr
+  && typeof tr.plural === 'string'
+  && typeof tr.singular === 'string';
+
+const localesKeys = ['en', 'nl'] as const;
```

Comment:
you could reuse 
type Locale = keyof Translations;
from locale.ts

Reply:
Well, not really, `Locale` is based on `Translations`. So it would be a circular dependency.

That being said, I exported `localeKeys` and `LocaleKeys`, which are both unused outside, so no need for this, and `LocaleKeys` is the same as `Locale` so I will delete it.